backend/dataall/cdkproxy/stacks/dataset.py

```diff
@@ -194,91 +194,114 @@ def __init__(self, scope, id, target_uri: str = None, **kwargs):
                         's3:GetAccessPointPolicyStatus',
-                        's3:DeleteAccessPointPolicy',
-                        's3:PutAccessPointPolicy',
                     ],
                     effect=iam.Effect.ALLOW,
                     resources=[
-                        f'arn:aws:s3:{dataset.region}:{dataset.AwsAccountId}:accesspoint/*',
+                        f'arn:aws:s3:{dataset.region}:{dataset.AwsAccountId}:accesspoint/{dataset.datasetUri}*',
                     ],
                 ),
                 iam.PolicyStatement(
-                    actions=['s3:List*', 's3:Get*'],
-                    resources=[f'arn:aws:s3:::{env.EnvironmentDefaultBucketName}'],
-                    effect=iam.Effect.ALLOW,
-                ),
-                iam.PolicyStatement(
-                    actions=['s3:*'],
-                    effect=iam.Effect.ALLOW,
-                    resources=[f'arn:aws:s3:::{env.EnvironmentDefaultBucketName}/*'],
-                ),
-                iam.PolicyStatement(
-                    effect=iam.Effect.ALLOW,
-                    resources=['arn:aws:s3:::aws-glue-*'],
-                    actions=['s3:CreateBucket'],
-                ),
-                iam.PolicyStatement(
-                    actions=['s3:GetObject', 's3:PutObject', 's3:DeleteObject'],
-                    effect=iam.Effect.ALLOW,
-                    resources=[
-                        'arn:aws:s3:::aws-glue-*/*',
-                        'arn:aws:s3:::*/*aws-glue-*/*',
+                    sid="GlueAccessCrawler",
+                    actions=[
+                        "glue:GetDatabase",
+                        "glue:GetTableVersion",
+                        "glue:CreateTable",
+                        "glue:GetTables",
+                        "glue:GetTableVersions",
+                        "glue:UpdateTable",
+                        "glue:DeleteTableVersion",
+                        "glue:DeleteTable",
+                        "glue:GetTable"
                     ],
-                ),
-                iam.PolicyStatement(
-                    actions=['s3:GetObject'],
                     effect=iam.Effect.ALLOW,
                     resources=[
-                        'arn:aws:s3:::crawler-public*',
-                        'arn:aws:s3:::aws-glue-*',
-                    ],
+                        f"arn:aws:glue:*:{dataset.AwsAccountId}:catalog",
+                        f"arn:aws:glue:{dataset.region}:{dataset.AwsAccountId}:database/{dataset.GlueDatabaseName}",
+                        f"arn:aws:glue:{dataset.region}:{dataset.AwsAccountId}:table/{dataset.GlueDatabaseName}/*"
+                    ]
                 ),
                 iam.PolicyStatement(
+                    sid="LoggingGlueCrawler",
                     actions=[
-                        'logs:CreateLogGroup',
-                        'logs:CreateLogStream',
                         'logs:PutLogEvents',
                     ],
```

Comment:
Confirming that on new dataset creation and first run of a new dataset crawler - does it not need `logs:CreateLogStream` permissions?

Reply:
Let me check it!

Reply:
Incredible catch! Thank you @noah-paige! The issue was that previously whenever we run the crawler the pivot role updated the crawler to use the pivotRole as execution role. So when I was testing I removed this behavior after the log group was created, damm

<img width="1159" alt="image" src="https://github.com/awslabs/aws-dataall/assets/71252798/5bd2ab8e-03b6-4358-8b0a-4876c3cc8371">

Reply:
I implemented and tested the permission problem in my last commit